symtabAPI/src/Object-elf.C

```diff
@@ -417,6 +418,11 @@ bool ObjectELF::loaded_elf(Offset &txtaddr, Offset &dataddr,
             foundInterp = true;
         } else if (elfPhdr.p_type() == PT_LOAD) {
             hasProgramLoad_ = true;
+        } else if (elfPhdr.p_type() == PT_RISCV_ATTRIBUTES) {
+            size_t len;
+            riscv_attrs.riscv_attr_string = std::string(
```

Comment:
This is not good. You're assigning `riscv_attrs.riscv_attr_string` to the entire ELF attribute data structure here. Then later you you replace this value with the contents of the `RiscvAttrTag::arch` tag's value (this is what the value should be).  Make two new member that are the pointer to the beginning of the data and it length (just like the other `*_addr_`, and `*_size_` members) instead of making a string and overloading this member.

Reply:
Makes sense.
Will do